core/shared/src/main/scala/zio/logging/internal/FileWriter.scala

```diff
@@ -48,7 +56,68 @@ private[logging] class FileWriter(
       writer.flush()
   }
 
-  final def flush(): Unit = writer.flush()
+  final def flush(): Unit = writerProvider.writer.flush()
+
+  final def close(): Unit = writerProvider.writer.close()
+
+  private[logging] sealed trait WriterProvider {
```

Comment:
now `WriterProvider` is inside `FileWriter`

i think you can move it out, so also it can be tested separately

Reply:
I wroted, Could you check this commit? https://github.com/zio/zio-logging/pull/691/commits/5dd2e73bdadd2d54abb16e9c9cf0296f4b71ba6d